### ISSUE TITLE
[BUG] Prot Warrior recommendations no longer working since PR3723

### ISSUE THREAD

**Issue by gbenbow** [Hekili/hekili#3730]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this issue (above) so that it describes the issue I am reporting.
- [X] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### Spec

Warrior - Protection

### Describe the Issue

Since the changes in https://github.com/Hekili/hekili/pull/3723 prot warrior recommendations are no longer functional (recommendations load but never change or react to ability usage). Reverting the changes in "WarriorProtection.lua" did fix the issue locally for me.

Looking into the Snapshot, I can see these warnings in relation to the changes made in the PR:
```
warnings: [#1] Returned unknown string 'hp' in state metatable [Protection Warrior:default:15].
[string "@Interface/AddOns/Hekili/State.lua"]:2389: in function <Interface/AddOns/Hekili/State.lua:2016>
[string "@Interface/AddOns/Hekili/TheWarWithin/WarriorProtection.lua"]:1282: in function `usable'
[string "@Interface/AddOns/Hekili/State.lua"]:7240: in function `IsUsable'
[string "@Interface/AddOns/Hekili/Core.lua"]:950: in function `GetPredictionFromAPL'
[string "@Interface/AddOns/Hekili/Core.lua"]:1414: in function `GetNextPrediction'
[string "@Interface/AddOns/Hekili/Core.lua"]:1793: in function <Interface/AddOns/Hekili/Core.lua:1464>
 
[#2] Update: ...ace/AddOns/Hekili/TheWarWithin/WarriorProtection.lua:1282: attempt to compare nil with number
[string "@Interface/AddOns/Hekili/TheWarWithin/WarriorProtection.lua"]:1282: in function `usable'
[string "@Interface/AddOns/Hekili/State.lua"]:7240: in function `IsUsable'
[string "@Interface/AddOns/Hekili/Core.lua"]:950: in function `GetPredictionFromAPL'
[string "@Interface/AddOns/Hekili/Core.lua"]:1414: in function `GetNextPrediction'
[string "@Interface/AddOns/Hekili/Core.lua"]:1793: in function <Interface/AddOns/Hekili/Core.lua:1464>
 
[#3] Returned unknown string 'hp' in state metatable [Protection Warrior:default:15].
[string "@Interface/AddOns/Hekili/State.lua"]:2389: in function <Interface/AddOns/Hekili/State.lua:2016>
[string "@Interface/AddOns/Hekili/TheWarWithin/WarriorProtection.lua"]:1282: in function `usable'
[string "@Interface/AddOns/Hekili/State.lua"]:7240: in function `IsUsable'
[string "@Interface/AddOns/Hekili/Core.lua"]:950: in function `GetPredictionFromAPL'
[string "@Interface/AddOns/Hekili/Core.lua"]:1414: in function `GetNextPrediction'
[string "@Interface/AddOns/Hekili/Core.lua"]:1793: in function `Update'
[string "@Interface/AddOns/Hekili/Options.lua"]:12107: in function `MakeSnapshot'
[string "@Interface/AddOns/Hekili/Options.lua"]:12064: in function `TogglePause'
[string "@Interface/AddOns/Hekili/Options.lua"]:12210: in function `FireToggle'
[string "@Interface/AddOns/Hekili/UI.lua"]:2853: in function <Interface/AddOns/Hekili/UI.lua:2852>
```

### How to Reproduce

1. Play as Prot Warrior
2. Pick any talents
3. Try to do rotation
4. See that Hekili doesn't update it's recommendations anymore

### Snapshot (Link)

https://pastebin.com/qVSvexdd

**Comment by johnnylam88** [comment]
This is fixed in #3729

**Comment by Hekili** [comment]
Fixed in 1.0.11b
